packages/daemon/src/mail.js

```diff
@@ -1,174 +1,112 @@
+      yield* subsequentRequests;
     };
 
     /**
-     * @param {object} partialMessage
-     * @returns {import('./types.js').InternalMessage}
+     * @param {import('./types.js').EnvelopedMessage} envelope
      */
-    const deliver = partialMessage => {
+    const deliver = envelope => {
       /** @type {import('@endo/promise-kit/src/types.js').PromiseKit<void>} */
       const dismissal = makePromiseKit();
       const messageNumber = nextMessageNumber;
       nextMessageNumber += 1;
 
+      const dismisser = makeExo(
+        'Dismisser',
+        M.interface(
+          'Dismisser',
+          {},
+          {
+            defaultGuards: 'passable',
+          },
+        ),
+        {
+          dismiss() {
+            messages.delete(messageNumber);
+            dismissal.resolve();
+          },
+        },
+      );
+
       const message = harden({
+        ...envelope,
         number: messageNumber,
-        when: new Date().toISOString(),
+        date: new Date().toISOString(),
         dismissed: dismissal.promise,
-        ...partialMessage,
-      });
-
-      dismissers.set(message, () => {
-        messages.delete(messageNumber);
-        dismissal.resolve();
+        dismisser,
       });
 
       messages.set(messageNumber, message);
       messagesTopic.publisher.next(message);
-
-      return message;
-    };
-
-    /**
-     * @param {string} what - user visible description of the desired value
-     * @param {string} who
-     * @param {string} dest
-     * @returns {Promise<string>}
-     */
-    const requestId = async (what, who, dest) => {
-      /** @type {import('@endo/promise-kit/src/types.js').PromiseKit<string>} */
-      const { promise, resolve } = makePromiseKit();
-      const settled = promise.then(
-        () => 'fulfilled',
-        () => 'rejected',
-      );
-      const message = deliver({
-        type: /** @type {const} */ ('request'),
-        who,
-        dest,
-        what,
-        settled,
-      });
-      resolvers.set(message, resolve);
-      return promise;
-    };
-
-    /** @type {import('./types.js').Mail['respond']} */
-    const respond = async (
-      what,
-      responseName,
-      senderId,
-      senderPetStore,
-      recipientId = selfId,
-    ) => {
-      if (responseName !== undefined) {
-        /** @type {string | undefined} */
-        let id = senderPetStore.identifyLocal(responseName);
-        if (id === undefined) {
-          id = await requestId(what, senderId, recipientId);
-          await senderPetStore.write(responseName, id);
-        }
-        // Behold, recursion:
-        // eslint-disable-next-line no-use-before-define
-        return provide(id);
-      }
-      // The reference is not named nor to be named.
-      const id = await requestId(what, senderId, recipientId);
-      // TODO:
-      // context.thisDiesIfThatDies(id);
-      // Behold, recursion:
-      // eslint-disable-next-line no-use-before-define
-      return provide(id);
```

Comment:
The fact that the recipient needed the sender’s pet store diffused a pretty awful smell. The pet store wasn’t even transmissible (not a Far/Exo). The `request` function uses `requestId` directly now and handles all the pet name preservation.